Use generic config parameters in pageserver cli (#1042)
zenithdb/zenith: docs/settings.md

## Pageserver

Pageserver is mainly configured via a `pageserver.toml` config file.

> If there's no such file durinig `init` phase of the server, it crates the file itself, otherwise it does not alter and parses it.

Reviewer: ```suggestion
If there's no such file during `init` phase of the server, it creates the file itself. Without 'init', the file is read.
```
This is a bit confusing, because the `pageserver.toml` file cannot exist when you create a new repository, because the `init` creates the directory.

There's a possibility to pass an arbitrary config value to the pageserver binary as an argument: such values override
the values in the config file, if any are specified for the same key and get into the final config during init phase.

### Config example

```toml
# Initial configuration file created by 'pageserver --init'

listen_pg_addr = '127.0.0.1:64000'
listen_http_addr = '127.0.0.1:9898'

checkpoint_distance = '268435456' # in bytes
checkpoint_period = '1 s'

gc_period = '100 s'
gc_horizon = '67108864'

> open_mem_limit = '134217728' # in bytes

Author: This parameter is not used anywhere, should we remote it?

Reviewer: yeah
